core/src/main/java/kafka/server/SharePartitionManager.java

```diff
@@ -129,23 +138,32 @@ private void maybeProcessFetchQueue() {
         Map<TopicIdPartition, FetchRequest.PartitionData> topicPartitionData = new LinkedHashMap<>();
         ShareFetchPartitionData shareFetchPartitionData = fetchQueue.poll();
         try {
-          assert shareFetchPartitionData != null;
-          shareFetchPartitionData.topicIdPartitions.forEach(topicIdPartition -> {
+            assert shareFetchPartitionData != null;
+            shareFetchPartitionData.topicIdPartitions.forEach(topicIdPartition -> {
+                SharePartitionKey sharePartitionKey = sharePartitionKey(
+                        shareFetchPartitionData.groupId,
+                        topicIdPartition
+                );
                 // TODO: Fetch inflight and delivery count from config.
-                SharePartition sharePartition = partitionCacheMap.computeIfAbsent(sharePartitionKey(
-                    shareFetchPartitionData.groupId, topicIdPartition),
-                    k -> new SharePartition(shareFetchPartitionData.groupId, topicIdPartition, 100, maxDeliveryCount,
-                            recordLockDurationMs, timer, time));
+                SharePartition sharePartition = partitionCacheMap.computeIfAbsent(sharePartitionKey,
+                    k -> new SharePartition(shareFetchPartitionData.groupId, topicIdPartition, maxDeliveryCount,
+                            maxInFlightMessages, recordLockDurationMs, timer, time));
                 int partitionMaxBytes = shareFetchPartitionData.partitionMaxBytes.getOrDefault(topicIdPartition, 0);
                 // Add the share partition to the list of partitions to be fetched only if we can
                 // acquire the fetch lock on it.
                 if (sharePartition.maybeAcquireFetchLock()) {
-                    topicPartitionData.put(topicIdPartition, new FetchRequest.PartitionData(
-                        topicIdPartition.topicId(),
-                        sharePartition.nextFetchOffset(),
-                        0,
-                        partitionMaxBytes,
-                        Optional.empty()));
+                    if (sharePartition.canAcquireMore()) {
+                        topicPartitionData.put(topicIdPartition, new FetchRequest.PartitionData(
+                                topicIdPartition.topicId(),
+                                sharePartition.nextFetchOffset(),
+                                0,
+                                partitionMaxBytes,
+                                Optional.empty()));
+                    } else {
+                        sharePartition.releaseFetchLock();
+                        log.trace("record lock partition limit exceeded for SharePartition with key {}, " +
```

Comment:
Shall it be INFO as something to track as why partition is stuck or atleast debug?